core/src/main/scala/hedgehog/state/Command.scala

```diff
@@ -45,6 +45,14 @@ trait Command[S, I, O] extends CommandIO[S] {
    */
   def ensure(env: Environment, before: S, after: S, i: Input, o: Output): Result
 
+  /**
+   * Render the input for displaying in test output.
+   *
+   * The default is to use `toString`, but optionally can support being overridden.
+   */
+  def renderInput(i: Input): String =
+    i.toString
```

Comment:
Is there any change that `i` can be `null`?

Reply:
It's _always_ possible on the JVM. :(

Fair point though. Have used `String.valueOf`

Reply:
Yeah, if so, I think `String.valueOf` is better. I had NPE when I used `toString` for logging in `diff`.